gaps/gap-4_comp_manifest/gap-4_comp_manifest.md

```diff
@@ -0,0 +1,271 @@
+---
+gap: 4
+title: Computation Manifest
+description:
+author: mf (@mfranciszkiewicz)
+status: Draft
+type: Feature
+---
+
+## Abstract
+
+This GAP introduces a manifest file specification and a set of GCRS (Golem
+Computation Resource Standards) properties (herein after referred to as
+"Computation Manifest"), which allow Requestors to declare the constraints of
+computations they may execute on a Provider's machine.
+
+## Motivation
+
+The need for a Computation Manifest originates from the upcoming Internet
+access feature, where each outbound request is routed via Provider's host
+operating system. In order to prevent unwanted / illegal behavior on Provider's
+behalf, she needs to remain in control of actions that Requestors can execute
+on her machine.
+
+With Computation Manifests, Requestors constrain themselves to
+a certain set of allowed actions, to be negotiated with and approved by a
+Provider. Requestors' actions will be verified against the Manifest during
+computation.
+
+## Specification
+
+Computation Manifest can take one of the following forms:
+
+- an embedded JSON / YAML object within an external configuration file,
+- `golem.srv.comp.manifest` namespace properties, an extension to
+  Golem Computation Resource Standards.
+
```

Comment:
I get the scenario with Computation Manifest included in the Demand - as it specifies the actions/commands the Requestor will want to run on the host.

What about the "external configuration file" scenario? What configuration file would that be? Where would it reside? Would it be included in some sort of a "Golem application metadata"? Is this a step towards a Golem application "gallery"?

Reply:
In most cases, the Computation Manifest will be embedded in the Payload Manifest, where CM is co-signed with image hashes. This becomes the basic, trusted way of using the image. 

Apart from that, we'd rarely expect for CM properties to appear in a Demand. If they do, they're meant to extend the original manifest. Providers may agree to these terms if their internal rule set allows, e.g., running `/bin/date` within images signed by authority `A`. Of course it's possible to specify the whole CM in the Demand and rely on Provider's internal rule set, because the CM is not signed by a trusted party.

The "external configuration file" is an exercise for developers extending the manifest and the tooling around it. Rephrasing, it should be possible to embed such manifest in some location specified in a Demand, so that the manifest could be extracted and verified